tests/unit/amo/components/TestSearchForm.js

```diff
@@ -1,179 +1,394 @@
+      });
+      expect(wrapper.find(Suggestion)).toHaveLength(0);
+      // this triggers Autosuggest
+      wrapper.find('input').simulate('focus');
+      expect(wrapper.find(Suggestion)).toHaveLength(2);
+      expect(wrapper.find(LoadingText)).toHaveLength(0);
+    });
 
-    const state = store.getState();
+    it('does not display suggestions when search is empty', () => {
+      const { store } = dispatchAutocompleteResults({ results: [
+        createFakeAutocompleteResult(),
+        createFakeAutocompleteResult(),
+      ] });
+      const { autocomplete: autocompleteState } = store.getState();
 
-    expect(mapStateToProps(state).addonType).toEqual(ADDON_TYPE_EXTENSION);
+      // setting the `query` prop to empty also sets the input state to empty.
+      wrapper = mountComponent({
+        query: '',
+        suggestions: autocompleteState.suggestions,
+      });
+      wrapper.find('input').simulate('focus');
+      expect(wrapper.find(Suggestion)).toHaveLength(0);
+    });
+
+    it('does not display suggestions when there is no suggestion', () => {
+      wrapper = mountComponent({ suggestions: [] });
+      wrapper.find('input').simulate('focus');
+      expect(wrapper.find(Suggestion)).toHaveLength(0);
+      expect(wrapper.find(LoadingText)).toHaveLength(0);
+    });
+
+    it('displays 10 loading bars when suggestions are loading', () => {
+      const { store } = dispatchSignInActions();
+
+      store.dispatch(autocompleteStart({
+        errorHandlerId: createStubErrorHandler().id,
+        filters: { query: 'test' },
+      }));
+
+      wrapper = mountComponent(mapStateToProps(store.getState()));
+      wrapper.find('input').simulate('focus');
+      expect(wrapper.find(LoadingText)).toHaveLength(10);
+    });
+
+    it('updates the state and push a new route when a suggestion is selected', () => {
+      const result = createFakeAutocompleteResult();
+      const { store } = dispatchAutocompleteResults({ results: [result] });
+      const { autocomplete: autocompleteState } = store.getState();
+
+      wrapper = mountComponent({
+        query: 'foo',
+        suggestions: autocompleteState.suggestions,
+      });
+      expect(wrapper.state('searchValue')).toEqual('foo');
+
+      wrapper.find('input').simulate('focus');
+      wrapper.find(Suggestion).simulate('click');
+      expect(wrapper.state('searchValue')).toEqual('');
+      sinon.assert.callCount(router.push, 1);
+      sinon.assert.calledWith(router.push, result.url);
+    });
+
+    it('ignores loading suggestions that are selected', () => {
+      const result = createFakeAutocompleteResult();
+      const { store } = dispatchAutocompleteResults({ results: [result] });
+      const { autocomplete: autocompleteState } = store.getState();
+
+      wrapper = mountComponent({ query: 'baz' });
+      expect(wrapper.state('searchValue')).toEqual('baz');
+
+      wrapper.instance().handleSuggestionSelected(createFakeEvent(), {
```

Comment:
did `wrapper.find(Suggestion).at(0).simulate('click')` not work for this?

Reply:
No because it is tricky to render `Suggestion` without actual suggestions. I mean, I cannot simulate a click on "nothing", I have to force the call to `handleSuggestionSelected()`.

Reply:
I cannot make it work 😢

Reply:
ok, no worries. It seemed worth a try!